src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -651,6 +652,9 @@ ReactDOMComponent.Mixin = {
             markup = DOMPropertyOperations.createMarkupForCustomAttribute(propKey, propValue);
           }
         } else {
+          if (__DEV__) {
+            ReactDOMInstrumentation.debugTool.onCreateMarkupForProperty(propKey, propValue, this._currentElement);
```

Comment:
Ah,  I didn't realize that operation was called directly. I've another
approach in mind as well,  I'll try it out next week.
On Apr 1, 2016 6:35 PM, "Jim" <notifications@github.com> wrote:

> In src/renderers/dom/shared/ReactDOMComponent.js
> <https://github.com/facebook/react/pull/6398#discussion_r58281313>:
>
> > @@ -651,6 +652,9 @@ ReactDOMComponent.Mixin = {
> >              markup = DOMPropertyOperations.createMarkupForCustomAttribute(propKey, propValue);
> >            }
> >          } else {
> > +          if (__DEV__) {
> > +            ReactDOMInstrumentation.debugTool.onCreateMarkupForProperty(propKey, propValue, this._currentElement);
>
> This line is the main one that concerns me. It seems like this call should
> be inside createMarkupForProperty such that it gets invoked whenever
> anyone calls it will get the onCreateMarkupForProperty event. However, we
> should not be passing the element to createMarkupForProperty.
>
> Thus, why this PR is difficult, thus my comments in #6062 (comment)
> <https://github.com/facebook/react/issues/6062#issuecomment-185528241>
> and #6062 (comment)
> <https://github.com/facebook/react/issues/6062#issuecomment-185535439>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/react/pull/6398/files/295b048adf2a1de62d2572c179da584bccfc7cb1#r58281313>
>